ibis/backends/clickhouse/tests/snapshots/test_select/test_named_from_filter_groupby/out1.sql

```diff
@@ -1,12 +1,17 @@
 SELECT
-  t0.key,
+  t1.key,
```

Comment:
Seems like we should have a SQL-like aggregate variant having the predicates encapsulated next to the aggregation groups and metrics, what do you think? I can add that in a follow-up if desired.

Reply:
Yeah, I think that could make sense. I wonder how difficult it would be to rolling that into `Select`. It might add a lot of complexity to do that. Either way, we can play around with the implementations.

Reply:
Created a follow-up https://github.com/ibis-project/ibis/issues/7851